tests/conftest.py

```
@@ -646,36 +635,42 @@ def test_something(classroom_context):
             student.seat          # Seat instance
             student.profile       # IdentityProfile instance
     """
-    from tests.helpers.context_factory import ClassroomContextFactory
+    from tests.helpers.classroom_initializer import initialize
 
     def _factory(**kwargs):
-        return ClassroomContextFactory(db, **kwargs).build()
+        classroom_key = kwargs.pop("classroom_key", "chemistry_p1")
+        classroom = initialize(classroom_key, app)
+        if kwargs:
+            raise TypeError(f"Unsupported classroom_context kwargs: {sorted(kwargs)}")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)